src/main/resources/user/walkthrough/Traversing.md

```diff
@@ -0,0 +1,112 @@
+
+The main Operations that can be used to traverse a graph are described below using this graph:
+
+```
+
+                 --> 7 <--
+               /           \
+              /             \
+             6  -->  3  -->  4
+ ___        ^         \
+|   |       /          /
+ -> 8 -->  5  <--  2  <
+          ^        ^
+         /        /
+    1 --         /
+     \          /
+       --------
+```
+
+## GetWalks(GetElements, GetElements)
+
+A GetWalks operation is used to retrieve all of the walks in a graph starting from one of a set of provided EntityIds, with a maximum length.
+
+A GetWalks operation is configured using a user-supplied list of GetElements operations. These are executed sequentially, with the output of one operation providing the input EntityIds for the next.
+
+A simple example of this can be seen below:
+
+${START_JAVA_CODE}
+${GET_WALKS_SIMPLE_SNIPPET}
+${JSON_CODE}
+${GET_WALKS_SIMPLE_JSON}
+${FULL_JSON_CODE}
+${GET_WALKS_SIMPLE_FULL_JSON}
+${PYTHON_CODE}
+${GET_WALKS_SIMPLE_PYTHON}
+${END_CODE}
+
+Here are the results:
+
+```
+${GET_WALKS_SIMPLE_RESULT}
+```
+
+For more details and for more detailed examples of the GetWalks operation please see [GetWalks](operations/getwalks.md)
+
+## GetElements -> ToVertices -> ToEntitySeeds -> GetElements
+
+It is valid to use GetElements -> GetElements to traverse a graph but you probably will not get the results you want.
+The first GetElements hop (starting at 1) would return Edges 1 -> 5 and 1 -> 2.  The second hop would return Entities at 1, 5, 1 and 2 
+followed by the Edges again 1 -> 5 and 1 -> 2.
+
+Therefore, in order to get GetElements and GetElements to chain together you need to use ToVertices(UseMatchedVertex=OPPOSITE) to extract 2 and 5, then ToEntitySeeds to convert it into new  and EntitySeed(2) EntitySeed(5) then you can do GetElements again.
+
+This example will be done in 4 stages shown below:
+
+GetElements to return Edges.
+Use ToVertices to extract the destination of the edges.
+Wrap the destination vertices in EntitySeeds.
+GetElements again.
+
+Sample code can be seen below:
+
+${START_JAVA_CODE}
+${TO_VERTICES_SIMPLE_SNIPPET}
+${JSON_CODE}
+${TO_VERTICES_SIMPLE_JSON}
+${FULL_JSON_CODE}
+${TO_VERTICES_SIMPLE_JSON}
+${PYTHON_CODE}
+${TO_VERTICES_SIMPLE_PYTHON}
+${END_CODE}
+
+Here are the results:
+
+```
+${TO_VERTICES_SIMPLE_RESULT}
+```
+
+For more details and for more detailed examples of the ToVertices operation please see [ToVertices](operations/tovertices.md)
+
+This however is quite wasteful and you would be better using GetAdjacentIds, shown below.  One reason it would be useful to use this chained method
+would be if you need to cache the results from the first hop, for example GetElements -> ExportToSet -> ToVertices -> ToEntitySeeds -> GetElements.
+
+## GetAdjacentIds
+
+A GetAdjacentIds operation will return the vertex at the opposite end of connected edges to a provided seed vertex.
```

Comment:
After that sentence, perhaps add:

This means you can easily chain multiple GetAdjacentIds operations together and you will simply traverse around the graph, performing one hop at a time.

If you wish to return the properties of the final Edges you traverse down you can replace the last GetAdjacentIds operation in your chain with a GetElements operation.